database/src/models.rs

```diff
@@ -11,12 +11,45 @@ use chrono::offset::Utc;
 use chrono_humanize::Humanize;
 use diesel;
 use diesel::prelude::*;
+use flaken::Flaken;
 use pwhash::bcrypt;
 use rocket::outcome::IntoOutcome;
 use rocket::request::{self, FromRequest, Request};
 use std::borrow::Cow;
 use {BASE_URL, DOMAIN};
 
+pub struct IdGenerator {
```

Comment:
Is this just so we could switch away from Flaken or add more state in the future?

Reply:
A bit of both.  aschmitz had good arguments for why the Mastodon ID generator was done the way it was, but putting it in the database would have made it a lot more difficult for my SQLite plans j(and flaken is immediately useful anyway).  Given this, I wanted an interface that could be messaged with "give me an ID", "give me the next ID", etc. without tying too much to any one implementation.

Reply:
Ahhh! :+1: